### ISSUE TITLE
upgrade from 1.4.2 to 1.5.6 is failing due to port already in use

### ISSUE THREAD

**Issue by ghost** [argoproj/argo-cd#5509]
We are trying to take a backup using 1.4.2 image but its not worked out as aws-cli not available in that image, forced to take backup using 1.5.6

Initiated upgrade and see all the pods related to argocd is failing to start except dex server, when we see the logs its stating that the port already in use. Tried to restart pods and scaled back the replicas to zero to see if that solves the problem, but it never.

**To Reproduce**

1. upgrade from 1.4.2 to 1.5.6 **command**: kubectl apply -n argocd -f https://raw.githubusercontent.com/argoproj/argo-cd/v1.5.6/manifests/install.yaml
2. all pods should be up and running, but we observed they are failed to start except dex  server pod

**Expected behavior**
upgrade pods should be running

**Version**

```shell
Paste the output from `argocd version` here.
```

**Logs**

```
$$$
k logs argocd-repo-server-fsdfsd-xrrwg -n argocd
error: a container name must be specified for pod argocd-repo-server-58bf5769cc-xrrwg, choose one of: [argocd-repo-server repo-server]
A312866@A312866Z-HFD-OEM hc-kafka-test % k logs argocd-repo-server-58bf5769cc-xrrwg -n argocd -c argocd-repo-server
time="2021-02-12T16:36:52Z" level=info msg="argocd-repo-server v1.5.6+f46333e serving on [::]:8081"
A312866@A312866Z-HFD-OEM hc-kafka-test % k logs argocd-repo-server-58bf5769cc-xrrwg -n argocd -c repo-server
time="2021-02-12T16:40:00Z" level=fatal msg="listen tcp :8081: bind: address already in use"

$$$
k logs argocd-application-controller-gffdgdfgfd-mcqq9 -n argocd -c application-controller
time="2021-02-12T16:40:09Z" level=info msg="appResyncPeriod=3m0s"
time="2021-02-12T16:40:09Z" level=info msg="Application Controller (version: v1.4.2+48cced9) starting (namespace: argocd)"
time="2021-02-12T16:40:09Z" level=info msg="Starting configmap/secret informers"
time="2021-02-12T16:40:09Z" level=fatal msg="listen tcp 0.0.0.0:8082: bind: address already in use"
```

NAME                                             READY   STATUS             RESTARTS   AGE
argocd-application-controller-gffdgdfgfd-mcqq9   1/2     CrashLoopBackOff   4          3m24s
argocd-dex-server-hghg-cwrms                2/2     Running            0          3m34s
argocd-redis-hgfhf-675f7                    1/2     CrashLoopBackOff   5          3m34s
argocd-redis-gdfgd-wl4pk                    1/1     Running            0          18h
argocd-repo-server-gfdgd-xrrwg              1/2     Error              5          3m34s
argocd-repo-server-gfdg-dfq8f              1/1     Running            0          17h
argocd-server-fsdfsd-6mhxd                   0/2     CrashLoopBackOff   1          15s
argocd-server-fdgdf-p9rs4                   1/1     Running            0          18h

**Comment by jannfis** [comment]
It seems that something went wrong when you ran `kubectl apply ...` and that your pods are spinning up two containers now, instead of one. Possibly the previous (1.4.2) and the new (1.5.6) one, at least your logs indicate that.

**Comment by kotlak** [comment]
Yes one is an old one and the other is a new version, the new version pods are getting port Is already in use which causing the upgrade to stuck and the old version keeps running at that moment. Now in this case we need to know why the port already in use error is coming, we did the following actions when this error came

1) Deleted new version pods that stuck in port already in use and different sequences
2) Tried to force upgrade to 1.6 even though 1.5 not upgraded successfully
3) finally, loading backup as part of disaster recovery, caused entire things to reset, ex: we see namespace deletion

none of the above three methods worked perfectly, i believe we should know which port or service need to be released(any manual intervention?) to get the new version pods running.

**Comment by Coder63** [comment]
Running into same issue . Do we have fix for this one ? I'm from migrating from v1.7.8 -> v1.8.0
